dvc/dependency/base.py

```diff
@@ -44,9 +44,6 @@ def update(self, rev=None):
         if self.fs.version_aware:
             self.fs_path = self.fs.path.version_path(self.fs_path, rev)
             self.meta = self.get_meta()
-            self.def_path = self.fs.path.version_path(
-                self.def_path, self.meta.version_id
-            )
```

Comment:
The current behavior for version aware import-url deps is that `def_path` should always be unversioned. `fs_path` is versioned (so update should only change  `meta.version_id` and `fs_path`).